convert2rhel/systeminfo.py

```diff
@@ -347,37 +349,47 @@ def get_enabled_rhel_repos(self):
         #     )
         return self.submgr_enabled_repos if not tool_opts.no_rhsm else tool_opts.enablerepo
 
-    def _check_internet_access(self, host="8.8.8.8", port=53, timeout=3):
+    def _check_internet_access(self):
         """Check whether or not the machine is connected to the internet.
 
-        This method will try to estabilish a socket connection through the
-        default host in the method signature (8.8.8.8). If it's connected
-        successfully, then we know that internet is accessible from the host.
+        This method will try to retrieve a web page on the Red Hat network that
+        we know to exist (http://static.redhat.com/test/rhel-networkmanager.txt).
+        If we can successfully access that page, then we decide we are connected
+        to the internet.
+
+        We check a web page because we will need working https to retrieve
+        packages from Red Hat infrastructure during the conversion.
 
         .. warnings::
             We might have some problems with this if the host machine is using
-            a NAT gateway to route the outbound requests
-
-        .. seealso::
-            * Comparison of different methods to check internet connections: https://stackoverflow.com/a/33117579
-            * Redirecting IP addresses: https://superuser.com/questions/954665/how-to-redirect-route-ip-address-to-another-ip-address
-
-        :param host: The host to establish a connection to. Defaults to "8.8.8.8".
-        :type host: str
-        :param port: The port for the connection. Defaults to 53.
-        :type port: int
-        :param timeout: The timeout in seconds for the connection. Defaults to 3.
-        :type port: int
-        :return: Return boolean indicating whether or not we have internet access.
+            a NAT gateway to route the outbound requests to any other service.
+
+            DNS could also be used to redirect the URL we test to another address.
+
+        :return: Return boolean indicating whether or not we have internet
+            access.
         :rtype: bool
         """
+        self.logger.info(
+            "Checking internet connectivity using address '%s'.",
+            CHECK_INTERNET_CONNECTION_ADDRESS,
+        )
         try:
-            self.logger.info("Checking internet connectivity using host '%s' and port '%s'." % (host, port))
-            socket.setdefaulttimeout(timeout)
-            socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect((host, port))
+            response = urllib.request.urlopen(CHECK_INTERNET_CONNECTION_ADDRESS)
+            response.close()
```

Comment:
I should have spotted this earlier... I wonder if we should also check the response code and body of this response.  If the user was behind some sort of captive portal, the internet check would not timeout but it would not return the red hat web site either.  So checking that the response body is `OK` feels like it is important.

Reply:
That becomes tricky. I brought it up before as that was the initial implementation - checking the content and the status code. But at that point you are not checking the internet connectivity, you're checking the website

If the website ever goes down with gateway error etc, we'd run into an issue. We should ignore response code. And as for the content, again if the website goes into a [503](https://http.cat/503) the content wouldn't be returned but the user still has internet - so checking for that means a reliance on it the service working as we expect it to work rather than just checking that the user has connectivity

Reply:
<nod> Makes sense.  Resolving this comment.